.github/workflows/build.yml

```diff
@@ -427,7 +431,8 @@ jobs:
           CC_VER=$( echo "${{ matrix.compiler }}" | awk '{ print $2; }')
           test "${CC_NAME}" = "clang" && CC_EXE="clang++"
           test "${CC_NAME}" = "gcc"   && CC_EXE="g++"
-          [[ "${{ matrix.compiler }}" = "GCC 8" ]] || EXTRA_CMAKE_FLAGS="$EXTRA_CMAKE_FLAGS -DPEDANTIC_COMPILER_WERROR=ON"
+          test "${{ matrix.compiler }}" = "GCC 8"  && EXTRA_CMAKE_FLAGS="$EXTRA_CMAKE_FLAGS -DPEDANTIC_COMPILER_WERROR=ON"
```

Comment:
> why did you decide on GCC 8 here to have pedantic warnings and warnings as errors set to on rather than a newer version of GCC or clang?

This is  a correction to the script since previously i believe logic was working incorrect, with || right side always evaluates